src/logger.service.ts

```diff
@@ -97,7 +99,7 @@ export class NGXLogger {
     const timestamp = new Date().toISOString();
     const config = this.configService.getConfig();
 
-    const callerDetails = NGXLoggerUtils.getCallerDetails();
+    const callerDetails = NGXLoggerUtils.getCallerDetails(customError);
```

Comment:
Just pass in message to getCallerDetails, and inside of that method, check to see if it's an instance of error.